api/v1beta1/cinderapi_types.go

```diff
@@ -18,93 +18,46 @@ package v1beta1
 
 import (
 	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CinderAPISpec defines the desired state of CinderAPI
-type CinderAPISpec struct {
-	// +kubebuilder:validation:Optional
-	// +kubebuilder:default=cinder
-	// ServiceUser - optional username used for this service to register in cinder
-	ServiceUser string `json:"serviceUser"`
-
-	// +kubebuilder:validation:Required
-	// ContainerImage - Cinder API Container Image URL (will be set to environmental default if empty)
-	ContainerImage string `json:"containerImage"`
+// CinderAPITemplate defines the input parameters for the Cinder API service
+type CinderAPITemplate struct {
```

Comment:
Adding this "middle layer" here allows to resolve the struct directly in the controller (`instance.Spec.CinderAPI` will include the `CinderServiceTemplate`, which collect the user input in the service CR).